kore/src/Kore/Internal/TermLike.hs

```diff
@@ -445,6 +446,33 @@ isFullySimplified = Attribute.isFullySimplified . extractAttributes
 simplifiedAttribute :: TermLike variable -> Pattern.Simplified
 simplifiedAttribute = Attribute.simplifiedAttribute . extractAttributes
 
+containsForall :: TermLike variable -> Bool
+containsForall = Recursive.cata containsForall0
```

Comment:
This is not what we need to be checking. We want to check that the free variables on the right-hand side of the claim (which are _implicitly_ universally quantified) also appear on the left-hand side of the claim. We don't need to walk the entire pattern, we just need to look at the `freeVariables`. If the right-hand side has an explicit `\forall` quantifier, I think we should assume that the user intended to write that, because by default the frontend won't allow it.